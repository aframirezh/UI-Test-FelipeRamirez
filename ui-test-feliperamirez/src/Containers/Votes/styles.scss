.container-votes {
    padding:40px;
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  
  .box-tittle-votes-size {
    font-size: 40px;
  }
  
  @media only screen and (max-width: 768px) {
    /* For mobile phones: */
    .container-votes  {
      grid-template-columns: 1fr;
    }
  }