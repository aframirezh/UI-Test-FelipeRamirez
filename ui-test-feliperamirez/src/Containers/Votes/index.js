import React, { Component } from 'react';
import './styles.scss';
import Card from '../Card';

const postulates = require("../../db.json");

var data = localStorage.getItem('postulates') == null ? postulates : JSON.parse(localStorage.getItem('postulates'));

class Votes extends Component {

    state = { profiles: data }
    render() {
        return (
            <div className='container-votes'>
                {this.state.profiles.map(profile => <Card key={profile.id} {...profile} />)}
            </div>
        );
    }
}

export default Votes;