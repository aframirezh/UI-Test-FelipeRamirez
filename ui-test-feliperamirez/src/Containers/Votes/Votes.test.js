import React from 'react';
import Votes from './';
import renderer from 'react-test-renderer';

describe('Votes rendering', () => {
    it('Votes correctly rendering', () => {
        const props = { id: 1 };
        const component = renderer.create(
            (<Votes {...props}/>),
        );

        const VotesGrid = component.toJSON();
        expect(VotesGrid).toMatchSnapshot();
 
    })
})