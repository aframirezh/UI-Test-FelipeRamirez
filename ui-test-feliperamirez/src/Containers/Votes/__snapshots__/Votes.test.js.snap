// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Votes rendering Votes correctly rendering 1`] = `
<div
  className="container-votes"
>
  <div
    className="card-container"
  >
    <div
      className="card-container_background card-container_background--1"
    >
      <div
        className="grid box"
      >
        <div>
          <section>
            <p>
              <strong
                className="box-tittle-votes-size regular"
              >
                Kanye West
              </strong>
            </p>
            <p
              className="thin"
            >
              1 month ago in Entertainment
            </p>
            <p
              className="regular"
            >
              Vestibulum diam ante, porttitor a odio eget, rhoncus neque.. Aenean eu velit libero
            </p>
            <div>
              <div
                className="middle"
              >
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="good"
                  />
                  <div
                    className="vote-good box-vote"
                  >
                    <span
                      className="oi oi-thumb-up"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="bad"
                  />
                  <div
                    className="good-bad box-vote"
                  >
                    <span
                      className="oi oi-thumb-down box-bad-down"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <button
                    className="button"
                    onClick={[Function]}
                  >
                    Vote now
                  </button>
                </label>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        className="grid box--good"
      >
        <span
          className="oi oi-thumb-up"
        />
      </div>
      <div
        className="grid box--bad"
      >
        <span
          className="oi oi-thumb-down box-bad-down"
        />
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="card-container_background card-container_background--2"
    >
      <div
        className="grid box"
      >
        <div>
          <section>
            <p>
              <strong
                className="box-tittle-votes-size regular"
              >
                Mark Zuckerberg
              </strong>
            </p>
            <p
              className="thin"
            >
              1 month ago in Bussines
            </p>
            <p
              className="regular"
            >
              Vestibulum diam ante, porttitor a odio eget, rhoncus neque.. Aenean eu velit libero
            </p>
            <div>
              <div
                className="middle"
              >
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="good"
                  />
                  <div
                    className="vote-good box-vote"
                  >
                    <span
                      className="oi oi-thumb-up"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="bad"
                  />
                  <div
                    className="good-bad box-vote"
                  >
                    <span
                      className="oi oi-thumb-down box-bad-down"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <button
                    className="button"
                    onClick={[Function]}
                  >
                    Vote now
                  </button>
                </label>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        className="grid box--good"
      >
        <span
          className="oi oi-thumb-up"
        />
      </div>
      <div
        className="grid box--bad"
      >
        <span
          className="oi oi-thumb-down box-bad-down"
        />
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="card-container_background card-container_background--3"
    >
      <div
        className="grid box"
      >
        <div>
          <section>
            <p>
              <strong
                className="box-tittle-votes-size regular"
              >
                Cristina Fernandez de Kirchner
              </strong>
            </p>
            <p
              className="thin"
            >
              1 month ago in Politics
            </p>
            <p
              className="regular"
            >
              Vestibulum diam ante, porttitor a odio eget, rhoncus neque.. Aenean eu velit libero
            </p>
            <div>
              <div
                className="middle"
              >
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="good"
                  />
                  <div
                    className="vote-good box-vote"
                  >
                    <span
                      className="oi oi-thumb-up"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="bad"
                  />
                  <div
                    className="good-bad box-vote"
                  >
                    <span
                      className="oi oi-thumb-down box-bad-down"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <button
                    className="button"
                    onClick={[Function]}
                  >
                    Vote now
                  </button>
                </label>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        className="grid box--good"
      >
        <span
          className="oi oi-thumb-up"
        />
      </div>
      <div
        className="grid box--bad"
      >
        <span
          className="oi oi-thumb-down box-bad-down"
        />
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="card-container_background card-container_background--4"
    >
      <div
        className="grid box"
      >
        <div>
          <section>
            <p>
              <strong
                className="box-tittle-votes-size regular"
              >
                Malala Yousafzai
              </strong>
            </p>
            <p
              className="thin"
            >
              1 month ago in Entertainment
            </p>
            <p
              className="regular"
            >
              Vestibulum diam ante, porttitor a odio eget, rhoncus neque.. Aenean eu velit libero
            </p>
            <div>
              <div
                className="middle"
              >
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="good"
                  />
                  <div
                    className="vote-good box-vote"
                  >
                    <span
                      className="oi oi-thumb-up"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <input
                    name="radio"
                    type="radio"
                    value="bad"
                  />
                  <div
                    className="good-bad box-vote"
                  >
                    <span
                      className="oi oi-thumb-down box-bad-down"
                    />
                  </div>
                </label>
                <label
                  className="order-left"
                >
                  <button
                    className="button"
                    onClick={[Function]}
                  >
                    Vote now
                  </button>
                </label>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        className="grid box--good"
      >
        <span
          className="oi oi-thumb-up"
        />
      </div>
      <div
        className="grid box--bad"
      >
        <span
          className="oi oi-thumb-down box-bad-down"
        />
      </div>
    </div>
  </div>
</div>
`;
